### PR TITLE
Update CHANGELOG for v0.5.0

### PR DESCRIPTION
**Pull request by tianyux-amz**
### Issue #, if available:
N/A

### Description of changes:
Update CHANGELOG for v0.5.0

### Testing done:
1. Tested on Xcode 11.4. The prebuilt SDK and Media binary works fine in the demo app.
2. Tested on Xcode 11.3. The prebuilt SDK and Media binary works fine in the demo app.
2. QA did sanity test on the demo app built with new binaries. No issue was found.

#### Manual test cases (add more as needed):

- [x] Join meeting
- [x] Leave meeting
- [x] Rejoin meeting
- [x] See metrics
- [x] See attendee presence
- [x] Send audio
- [x] Receive audio
- [x] Mute self
- [x] Unmute self
- [x] See local mute indicator when muted
- [x] See remote mute indicator when other is muted
- [x] Enable and disable local video
- [x] Enable and disable remote video from remote side
- [x] Send local video
- [x] Pause and resume remote video
- [x] Switch local camera
- [x] Receive remote screen share

#### Integration validation (required before release)

- [x] Unit tests pass
- [x] Build SDK against simulator with release options
- [x] Build SDK against device with release options
- [x] Build SDK against simulator with debug options
- [x] Build SDK against device with debug options
- [x] Test Demo against simulator with SDK (debug build) in workspace
- [x] Test Demo against device with SDK (debug build) in workspace
- [x] Test DemoObjC against simulator with SDK (debug build) in workspace
- [x] Test DemoObjC against device with SDK (debug build) in workspace
- [x] Test Demo against simulator with SDK.framework (release build)
- [x] Test Demo against device with SDK.framework (release build)
- [x] Test DemoObjC against simulator with SDK.framework (release build)
- [x] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:
N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1 +1,7 @@
+## [0.5.0] - 2020-03-27
 
+### Changed
+
```
Comment (linsang21) [inline CHANGELOG.md]: Extra line?

Comment (hokyungh) [reply CHANGELOG.md]: Will https://github.com/aws-amplify/aws-sdk-ios/issues/2365 be same issue and can be solved by setting distribution flag?

Comment (tianyux-amz) [reply CHANGELOG.md]: I don't think it matters because `-` will start a new block anyway, but I can remove this line to align with other CHANGELOGs.

**Comment by tianyux-amz** [comment]
Thanks @hokyungh for pointing me to aws-amplify/aws-sdk-ios#2365! I just did some testing and the `BUILD_LIBRARY_FOR_DISTRIBUTION` option works as expected. I will make the code change accordingly.

**Pull request merged**